utils/pattern/src/common.rs

```diff
@@ -111,6 +111,10 @@ pub trait PatternBackend: crate::private::Sealed + 'static + core::fmt::Debug {
     /// Iterates over the pattern items in a store.
     #[doc(hidden)] // TODO(#4467): Should be internal
     fn iter_items(store: &Self::Store) -> Self::Iter<'_>;
+
+    /// Writes the placeholder into the pattern string, the inverse of `Self::PlaceholderKey::from_str`
+    #[cfg(feature = "alloc")]
+    fn escape_placeholder(s: &mut String, placeholder: Self::PlaceholderKey<'_>);
```

Comment:
Thought: Ok, you added a trait function to do this. This seems like a minimal one. It does make some assumptions about the serialized string: the placeholders/literals are in order, and the literals don't need to be escaped. That second part is not true for datetime, for example (and it is why the pattern parser has the quoting flag). However, I don't plan on migrating datetime patterns to this type.

Reply:
This round trips correctly through serde. Whether an escaped pattern from CLDR JSON can be parsed as a pattern is a different story.

Reply:
You call `try_from_str` in `serde::Deserialize`, so it needs to output patterns in the same syntax as they are being parsed from a place like CLDR JSON

Reply:
Only if you're using the pattern type to parse CLDR JSON. If you want non-default parsing behaviour, you use a `String` in the CLDR serde and parse it yourself. The point it that patterns *serialize* without quotes, so in ICU4X JSON they roundtrip.

Reply:
Observation: a literal is allowed to contain `"{0}"` or other text that might "look like" a placeholder to the parsing step.

Reply:
reverted